src/lib.rs

```diff
@@ -854,6 +876,21 @@ impl<T, const N: usize> FromFn<T> for [T; N] {
     {
         core::array::from_fn(cb)
     }
+
+    fn try_from_fn<E, F>(mut cb: F) -> Result<Self, E>
+    where
+        F: FnMut(usize) -> Result<T, E>,
+    {
+        // SAFETY: every element of the array below is initialized before use.
+        let mut array: Self = unsafe { mem::zeroed() };
```

Comment:
This does not look safe at all. You do not constrain `T` in any way, so zero bit pattern may not be valid for it. Also, this function does not look safe in regards to potential panics in the closure. Imagine a type which executes some `Drop` action (e.g. if `T` is `Vec<u8>`) and then the closure panics on first call, the code unwinds and you execute `Drop` for a bunch of zeroed values in `array`, which is obviously UB.